### DISCUSSION TITLE
Add back populating for document

### DISCUSSION THREAD

**Discussion by ErezAmihud** [BeanieODM/beanie#515] [Feature Request]
In mongodb and other document based dbs one to many relationship presents itself as an additional attribute in the "many" part of the relationship.

For example, describing a user and a post relationship can look like:
```python
class User(Document):
    email: str

class Post(Document):
    user: Link[User]
```

When using this program, I might want to get all the posts a user has created, so I would like to have something like:
```python
class User(Document):
    email:str
    posts:Backlink[Post, Post.user]
```

When using it, the user posts attribute would not be saved in the user document but loaded from query all posts where `Post.user` if the current user.

tl;dr same as relationship(back_populates) in sqlalchemy, or even better, read [this section](https://mongoosejs.com/docs/6.x/docs/populate.html#populate-virtuals) about populating virtuals in mongoose

**Comment by roman-right** [comment]
Hi,
I've started implementing backrefs. Here is the branch. It looks like fetching works well already. I'll add other operations soon.

https://github.com/roman-right/beanie/tree/feat/back-refs

https://github.com/roman-right/beanie/blob/feat/back-refs/tests/odm/test_relations.py#L458
